projects/plugins/boost/app/features/guide/src/index.ts

```diff
@@ -0,0 +1,42 @@
+import { attachGuides } from './Attach';
+import { createImageObserver } from './ImageObserver';
+import { load } from './Images';
+import { measure } from './Measurements';
+import AdminBarToggle from './ui/AdminBarToggle.svelte';
+
+/**
+ * Initialize the admin bar toggle.
+ */
+document.addEventListener( 'DOMContentLoaded', () => {
+	const adminBarToggle = document.getElementById( 'wp-admin-bar-jetpack-boost-image-guide' );
+	const link = adminBarToggle?.querySelector( 'a' );
+	if ( adminBarToggle && link ) {
+		const href = link.getAttribute( 'href' );
+		link.remove();
+		new AdminBarToggle( {
+			target: adminBarToggle,
+			props: {
+				href,
+			},
+		} );
+	}
+} );
+
+/**
+ * Initialize the guides when window is loaded.
+ */
+window.addEventListener( 'load', async () => {
+	const nodes = document.querySelectorAll( 'body *' );
+	const images = await load( Array.from( nodes ) );
+	const measuredImages = measure( images );
+	attachGuides( measuredImages );
+} );
+
+/**
+ * Watch for new images.
+ */
+createImageObserver( async ( nodes: Element[] ) => {
+	const images = await load( nodes );
+	const measuredImages = measure( images );
+	attachGuides( measuredImages );
+} );
```

Comment:
We shouldn't be missing anything with the Observer in place, but since the observer is a bit wobbly at the moment, maybe it's worth replacing that with a manual scan on click 🤔

Reply:
How expensive is re-scanning the DOM for images? Unless it's super slow and clunky, I think Adnan's idea is solid.

Reply:
For clicks - not that expensive. I'm not entirely convinced in terms of usability - ideally we want 0 clicks, but as I said, while the MutationObserver isn't in the best shape - this is the second best option.

Reply:
Done, it now tracks state and initializes any new images found.